app/controllers/lab_tests_controller.rb

```diff
@@ -80,20 +88,55 @@ def destroy
 
   private
 
-  # Use callbacks to share common setup or constraints between actions.
+  def handle_success_response
+    flash[:notice] = t('.success')
+    redirect_to(@health_record)
+  end
+
+  def handle_error_response(message = nil)
+    flash.now[:alert] = message if message
+    load_error_dependencies
+    render :new, status: :unprocessable_entity
+  end
+
+  def determine_redirect_path
+    @health_record || @lab_test
+  end
+
+  def load_error_dependencies
+    return if @lab_test.biomarker_id.blank?
+
+    @selected_biomarker = Biomarker.find(@lab_test.biomarker_id)
+    @reference_ranges = @selected_biomarker.reference_ranges
+  end
+
+  def set_biomarkers
+    @biomarkers = Biomarker.all
+  end
+
   def set_lab_test
     @lab_test = LabTest.find(params[:id])
   end
 
-  def build_lab_test
-    @lab_test = current_user.lab_tests.build(lab_test_params)
+  def save_records
```

Comment:
The name of this method doesn't show the exact sense of what is going on inside.
What exact records are we saving?
How to determine from the name, that both `health_record` and `lab_test` records suppose to be saved?

I'm not quite sure if we even need to extract this functionality to a separate method.
If you still want, than please consider renaming it to display the exact functionality behind this method.